mqtt/mqtt-bridge/src/persist/publication_store.rs

```diff
@@ -6,22 +6,55 @@ use parking_lot::Mutex;
 use tracing::debug;
 
 use crate::persist::{
-    loader::MessageLoader, waking_state::StreamWakeableState, Key, PersistError, WakingMemoryStore,
+    loader::MessageLoader,
+    waking_state::StreamWakeableState,
+    Key,
+    PersistError, //WakingMemoryStore,
 };
 
+use super::storage::{ring_buffer::RingBuffer /*, sled::Sled*/, FlushOptions};
+
 /// Pattern allows for the wrapping `PublicationStore` to be cloned and have non mutable methods
 /// This facilitates sharing between multiple futures in a single threaded environment
 struct PublicationStoreInner<S> {
-    state: Arc<Mutex<S>>,
+    state: Arc<S>,
     offset: u64,
     loader: MessageLoader<S>,
 }
 /// Persistence implementation used for the bridge
 pub struct PublicationStore<S>(Arc<Mutex<PublicationStoreInner<S>>>);
 
-impl PublicationStore<WakingMemoryStore> {
-    pub fn new_memory(batch_size: usize) -> PublicationStore<WakingMemoryStore> {
-        Self::new(WakingMemoryStore::default(), batch_size)
+unsafe impl<S> Send for PublicationStore<S> {}
+
+// impl PublicationStore<WakingMemoryStore> {
+//     pub fn new_memory(batch_size: usize) -> PublicationStore<WakingMemoryStore> {
+//         Self::new(WakingMemoryStore::default(), batch_size)
+//     }
+// }
+
+// TODO: Sled
+// impl PublicationStore<Sled> {
+//     pub fn new_db(
+//         path: String,
+//         tree_name: String,
+//         flush_options: FlushOptions,
+//         batch_size: usize,
+//     ) -> Self {
+//         Self::new(Sled::new(path, tree_name, flush_options), batch_size)
+//     }
+// }
```

Comment:
I have commented out both Sled and Memory. My plan is to remove memory altogether, and once I have tests working and if there is some extra time I will try to add sled as an alternative option.